python/cudf/cudf/tests/test_s3.py

```diff
@@ -62,6 +62,9 @@ def s3_base(endpoint_ip, endpoint_port):
         os.environ.setdefault("AWS_ACCESS_KEY_ID", "foobar_key")
```

Comment:
Should these be “setdefault” or should they be overriding the existing environment variable if present?

Reply:
The moto examples are overriding the environment explicitly. https://docs.getmoto.org/en/latest/docs/getting_started.html#recommended-usage

Reply:
Since this would be just for pytest session scope, I think it's _okay_ to just mock them here..

Reply:
What I’m saying is that we shouldn’t ever send real credentials from the environment to the mocked S3 backend that moto is providing. Is that accurate?

Reply:
I tend to agree with this, but I don't know the history of these tests. I wasn't sure if `setdefault` was used for a particular reason.

@bdice, should I push an update to match their docs?

Reply:
Yes please. If tests pass, I think this is a better choice.

Reply:
Pushed in f602511.